components/speedreader/rust/lib/src/readability/src/dom.rs

```diff
@@ -134,54 +136,61 @@ pub fn extract_text(handle: &Handle, text: &mut String, deep: bool) {
 
 pub fn text_len(handle: &Handle) -> usize {
     let mut len = 0;
-    for child in handle.children.borrow().iter() {
-        match child.data {
-            Text { ref contents } => {
+    for child in handle.children() {
+        match child.data() {
+            Text(ref contents) => {
                 len += contents.borrow().trim().chars().count();
             }
-            Element { .. } => {
-                len += text_len(child);
+            Element(_) => {
+                len += text_len(&child);
             }
             _ => (),
         }
     }
     len
 }
 
-pub fn find_node(handle: &Handle, tag_name: &str, nodes: &mut Vec<Rc<Node>>) {
-    for child in handle.children.borrow().iter() {
-        if let Element { ref name, .. } = child.data {
-            let t = name.local.as_ref();
+#[inline]
+pub fn find_nodes_with_tag(handle: &Handle, tags: &[&str], nodes: &mut Vec<Handle>) {
```

Comment:
Any special reason for having this signature

```rust
pub fn find_nodes_with_tag(handle: &Handle, tags: &[&str], nodes: &mut Vec<Handle>) {}
```

instead of ?

```rust
pub fn find_nodes_with_tag(handle: &Handle, tags: &[&str]) -> Vec<Handle> {}

```

nit, but IMO returning a new object feels more readable and the API is easier to use.